Testscripts/Linux/XDP-Action.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+# This script verifies XDP Action - TX or DROP
+# For DROP Action verification script will compare number of ping packets
+#		lossed before XDP DROP action and after XDP DROP action
+# For TX Action verification script will start ping on server and measure the packets captured by tcpdump
+#		Expected result: 0 packets should be captured by tcpdump after TX action as this action
+#		will return all ping packets to sender at hardware level
+
+
+# Helper function
+# This function will start xdpdump on client
+# and ping from server on eth1 network interface
+function ping_test () {
+	local logSuffix=$1
+
+	xdpdumpCommand="cd bpf-samples/xdpdump && timeout 20 ./xdpdump -i ${nicName} > ~/xdpdumpout_${logSuffix}.txt"
+	LogMsg "Starting xdpdump on ${client} with command: ${xdpdumpCommand}"
+	ssh ${client} "sh -c '${xdpdumpCommand} &'"
+
+	LogMsg "Starting ping on ${server} with command: ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+	ssh ${server} "ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+	check_exit_status "ping test ${logSuffix} XDPDump-${ACTION} config"
+}
+
+# Helper function
+# This function will start tcpdump on client and start ping_test
+function ping_with_tcpdump () {
+	local logSuffix=$1
+	LogMsg "Ips client $client server $server"
+	# Start tcpdump
+	LogMsg "Starting tcpdump on ${client} with command timeout 10 tcpdump -i eth1 icmp -w tcpdumptest_${logSuffix}.pcap"
+	ssh ${client} "sh -c 'timeout 10 tcpdump -i eth1 icmp -w tcpdumptest_${logSuffix}.pcap &'"
+	ping_test $logSuffix
+}
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+		echo "ERROR: unable to source ${UTIL_FILE}!"
+		echo "TestAborted" > state.txt
+		exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Script start from here
+LogMsg "*********INFO: Script Execution Started********"
+
+# check for client and server ips are present
+for ip in ${client} ${server}
+do
+	CheckIP ${ip}
+	if [ $? -eq 1 ]; then
+		LogErr "ERROR: Please provide valide client and server ip. Invalide ip: ${ip}"
+		SetTestStateAborted
+		exit 0
+	fi
+done
+
+beforeString="before"
+afterString="after"
+packetLossInNetwork=10
+
+if [ "${ACTION}" == "DROP" ];then
+	LogMsg "Initializing validation for XDP Action ${ACTION}"
+
+	# Ping test before changing action
+	ping_test $beforeString
+
+	# Build xdpdump with DROP Config
+	ssh ${client} "cd bpf-samples/xdpdump && make clean && export CFLAGS='-D __ACTION_DROP__ -I../libbpf/src/root/usr/include' && make"
```

Comment:
export & and(&&) is not required.
Just do CFLAGS='-D__XXXXX_ -IXXXX' make

Reply:
export is required. In make file there is a check on CFLAGS is defined or not which only detects CFLAGS value if It is present in the environment. Please let me know if I'm wrong here I don't have an experience with make file that much but this is what I have observed.

Reply:
Fixed